res/controllers/Numark_DJ2GO2_Touch_scripts.js

```diff
@@ -0,0 +1,210 @@
+// Declare the variable for your controller and assign it to an empty object
+var DJ2GO2Touch = {};
+
+// Mixxx calls this function on startup or when the controller
+// is enabled in the Mixxx Preferences
+DJ2GO2Touch.init = function() {
+    // create an instance of your custom Deck object for each side of your controller
+    DJ2GO2Touch.leftDeck = new DJ2GO2Touch.Deck([1], 0);
+    DJ2GO2Touch.rightDeck = new DJ2GO2Touch.Deck([2], 1);
+};
+
+DJ2GO2Touch.shutdown = function() {
+    // send whatever MIDI messages you need to turn off the lights of your controller
+};
+
+DJ2GO2Touch.browseEncoder = new components.Encoder({
+    longPressTimer: 0,
+    longPressTimeout: 250,
+    previewSeekEnabled: false,
+    previewSeekHappened: false,
+    onKnobEvent: function(rotateValue) {
+        if (rotateValue !== 0) {
+            if (this.previewSeekEnabled) {
+                var oldPos = engine.getValue("[PreviewDeck1]", "playposition");
+                var newPos = Math.max(0, oldPos + (0.05 * rotateValue));
+                engine.setValue("[PreviewDeck1]", "playposition", newPos);
+            } else {
+                engine.setValue("[Playlist]", "SelectTrackKnob", rotateValue);
+            }
+        }
+    },
+    onButtonEvent: function(value) {
+        if (value) {
+            this.isLongPressed = false;
+            this.longPressTimer = engine.beginTimer(
+                this.longPressTimeout,
+                function() { this.isLongPressed = true; },
```

Comment:
This functionality doesn't seem to work, I copied it from the 2.3 branch of the DJ-505 mapping but couldn't figure it out, clicking the browser encoder knob breaks the scroll functionality

Reply:
its because `this` is actually different inside the function. The DJ-505 mapping avoids this using the `bind` call, but that is not available here. A workaround would be to alias `this` (using `var that = this` outside the handler and then access `that` instead of `this`. 
Alternatively, you can rethink whether this functionality is really required... this code is quite complex so if you don't need the functionality, it would be worth thinking about getting rid of unnecessary features to simplify the code.